File: iosApp/iosApp/Presentation/Main/CheckIn/CheckInUiState.swift
//
//  CheckInUiState.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/06.
//

import Foundation
import Shared

struct CheckInUiState {
    var isLoading: Bool
    var errorMessage: String?
    var contentLayout: AppContentLayout = .fullBleed
    var currentUser: User?
    var currentLocationLabel: String
    var userCityKey: String?
    var mapCafes: [CheckInCafeSummary]
    var popularCafes: [CheckInCafeSummary]
    var popularCasts: [CheckInCastSummary]
    var todayVisits: [CheckInVisitEntry]
    var recentVisits: [CheckInVisitEntry]
    var recentVisitsNextCursor: String?
    var canLoadMoreRecentVisits: Bool
    var isLoadingMoreRecentVisits: Bool
    var isLoginPromptVisible: Bool
    var loginPromptType: LoginPromptType
    var isNewVisitSheetVisible: Bool
    var isQrCheckInSheetVisible: Bool
    var preselectCafeId: String?
    var reviewPrompt: ReviewPrompt?
    var selectedMapRegion: ExploreUiState.RegionFilter

    static let empty = CheckInUiState(
        isLoading: false,
        errorMessage: nil,
        contentLayout: .fullBleed,
        currentUser: nil,
        currentLocationLabel: "",
        userCityKey: nil,
        mapCafes: [],
        popularCafes: [],
        popularCasts: [],
        todayVisits: [],
        recentVisits: [],
        recentVisitsNextCursor: nil,
        canLoadMoreRecentVisits: false,
        isLoadingMoreRecentVisits: false,
        isLoginPromptVisible: false,
        loginPromptType: .checkIn,
        isNewVisitSheetVisible: false,
        isQrCheckInSheetVisible: false,
        preselectCafeId: nil,
        reviewPrompt: nil,
        selectedMapRegion: .all
    )

    enum LoginPromptType {
        case detail
        case checkIn
    }

    struct ReviewPrompt {
        let visitId: String
        let cafeId: String
        let cafeName: String
    }
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/checkin/CheckInUiState.kt
package com.hhp227.concafe.presentation.main.checkin

import com.hhp227.concafe.domain.model.CheckInCafeSummary
import com.hhp227.concafe.domain.model.CheckInCastSummary
import com.hhp227.concafe.domain.model.CheckInVisitEntry
import com.hhp227.concafe.domain.model.User

data class CheckInUiState(
    val isLoading: Boolean = false,
    val errorMessage: String? = null,
    val currentUser: User? = null,
    val currentLocationLabel: String = "",
    val mapCafes: List<CheckInCafeSummary> = emptyList(),
    val popularCafes: List<CheckInCafeSummary> = emptyList(),
    val popularCasts: List<CheckInCastSummary> = emptyList(),
    val todayVisits: List<CheckInVisitEntry> = emptyList(),
    val recentVisits: List<CheckInVisitEntry> = emptyList(),
    val recentVisitsNextCursor: String? = null,
    val canLoadMoreRecentVisits: Boolean = false,
    val isLoadingMoreRecentVisits: Boolean = false,
    val isLoginPromptVisible: Boolean = false,
    val isNewVisitSheetVisible: Boolean = false,
    val reviewPrompt: ReviewPrompt? = null
) {
    data class ReviewPrompt(
        val visitId: String,
        val cafeId: String,
        val cafeName: String
    )

    companion object {
        fun empty() = CheckInUiState()
    }
}
